lib/import/admin.js

```diff
@@ -142,17 +150,57 @@ function save_features(feature_collection) {
 }
 
 /**
+ * Get hash of admin to population
+ * mutates `feature_collection`.
+ *
+ * @param{string} country_code - Country Code
+ * @param{array} feature_collection - Feature collection
+ * @return{Promise} Fulfilled when csv is read.
+ */
+function world_pop(country_code, feature_collection) {
+  var path = './data/population/';
+  var file = 'worldpop_' + country_code + '.csv';
+  var stream = fs.createReadStream(path + file);
+  var admin_pop = {};
+  return new Promise(function(resolve) {
+    stream.on('error', function(err) {
+      console.log('No population file. Moving along.', err);
+      resolve([{}, feature_collection]);
+    });
+
+    var csvStream = csv()
+    .on('data', function(data) {
+      admin_pop[data[0]] = parseFloat(data[1]);
+    })
+    .on('end', function() {
+      resolve([admin_pop, feature_collection]);
+    });
+    stream.pipe(csvStream);
+  });
+}
+
+/**
  * Import admins from a geojson file and save as Admin documents.
  *
- * @param{string} country_code - 3166-1 alpha-2 country code.
  * @param{string} file - name of geojson file.
  * @param{string} verbose - Option to display debug
  * @return{Promise} Fulfilled when documents are saved.
  */
-function import_admins(country_code, file, verbose) {
+function import_admins(file, verbose) {
+  var country_code = file.split('/')[file.split('/').length - 2];
   return read_jsonfile(file, verbose)
   .then(function(feature_collection) {
-    return enrich_features(country_code, feature_collection, verbose);
+    return world_pop(country_code, feature_collection);
+  })
+  .then(function(pop_feat_collection) {
+    var pop_admin = pop_feat_collection[0];
+    var feature_collection = pop_feat_collection[1];
```

Comment:
`world_pop` doesn't actually depend on or change `feature_collection`. I see that you just threaded it through so that `enrich_features` would have access to it. It would be better to restructure this code, though, so you don't have to do the unnecessary argument passing.

You should be able to do something like this instead:
```
  return Promise.all([read_jsonfile(file, verbose), world_pop(country_code)])
  .then(function(feat_collection_and_admin_pop) {
    var feature_collection = feat_collection_and_admin_pop[0];
    var pop_admin = feat_collection_and_admin_pop[1];
    return enrich_features(country_code, feature_collection, pop_admin,
                           verbose);
  })
  .then(function(feature_collection) {
    return save_features(feature_collection);
  })
  .then(function(feature_collection) {
    return save_topojson(country_code, feature_collection, file, verbose);
  });
```
(This also requires changing `world_pop` to no longer take or return the `feature_collection`.)

Now that I look at it, there's no reason for `enrich_features` to return a Promise at all... but let's fix that in a separate commit.